### ISSUE TITLE
Relocate functions to clarify API

### ISSUE THREAD

**Issue by kotfu** [python-cmd2/cmd2#878]
While working on the API documentation, I noticed that `decorators.py` contains two functions `categorize()` and `set_parser_prog()`. Shouldn't these functions be in `utils.py` instead? Is there a good reason why they are in `decorators.py`?

**Comment by kmvanbrunt** [comment]
They are related to decorated command functions.

**Comment by kotfu** [comment]
`categorize()` is imported by `__init__.py` so it could reasonably be moved without breaking anything. However, `set_parser_prog()` is not, so if we move it it will break people's imports. Perhaps I'll rename this issue and keep a running list of functions that could potentially be relocated and we'll see how long the list is before deciding whether to move them or not.

**Comment by kotfu** [comment]
## Functions/consts/classes to be relocated
Relocated means moved from one source file to another. In some cases this will break imports. This list tracks each proposed move, and whether it will break imports or not.

- `set_parser_prog()` decorators.py -> utils.py : breaks imports
- `categorize()` decorators.py -> utils.py : imported by `__init__.py` so no break

**Comment by kmvanbrunt** [comment]
I recently wrote `set_parser_prog()` as a support function for the `argparse` decorators. I don't think anyone calls them directly.

**Comment by kotfu** [comment]
So maybe it shouldn't be included in the public API?

**Comment by kmvanbrunt** [comment]
I'd be ok prepending an underscore.

**Comment by kmvanbrunt** [comment]
`set_parser_prog()'s` purpose is to solve the problem of `prog` getting properly set on subcommand's subparsers. `cmd2` doesn't require developers to set `prog` on parsers used in the `argparse` decorators since we set it to match the command's name for them. When you don't set `prog`, it defaults to `sys.argv[0]`. Each subparser you add inherits this name. Since our decorators were updating `prog` after all the subparsers had been added, they weren't picking up the change. Therefore usage text on subcommands would start with `sys.argv[0]`. The only way to fix this before was to set `prog` explicitly on the root parser. Most people didn't know this caveat.

To make things easier, I made the decorators do all the work by calling `set_parser_prog()`.
